src/lib/verifier/common.mli

```diff
@@ -0,0 +1,156 @@
+    both [User_command.Verifiable.t] (in-process) and
+    [User_command.Verifiable.Serializable.t] (cross-process via RPC). *)
+
+(** A zkApp command with verification key data attached.
+
+    {b Type parameters:}
+    - ['proof] is the proof type in account update authorizations
+    - ['aux] is auxiliary data attached to account updates *)
+type ('proof, 'aux) verifiable_zkapp_command =
+  ( ( Account_update.Body.t
+    , ('proof, Signature_lib.Schnorr.Chunked.Signature.t) Control.Poly.t
+    , 'aux )
+    Account_update.Poly.t
+  , Verification_key_wire.t option )
+  Zkapp_command.Call_forest.With_hashes_and_data.t
+  Zkapp_command.Poly.t
+
+(** A user command (signed or zkApp) in verifiable form.
+
+    {b Type parameters:}
+    - ['proof] is [Proof_cache_tag.t] for [User_command.Verifiable.t]
+      or [Pickles.Side_loaded.Proof.t] for [User_command.Verifiable.Serializable.t]
+    - ['aux] is [Account_update.T.Aux_data.t] for [User_command.Verifiable.t]
+      or [unit] for [User_command.Verifiable.Serializable.t] *)
+type ('proof, 'aux) verifiable_user_command =
+  ( Signed_command.t
+  , ('proof, 'aux) verifiable_zkapp_command )
+  User_command.Poly.t
+
+(** Verify a signed command's signature.
+
+    @return [Ok (`Assuming [])] if valid, [Error invalid] otherwise *)
+val check_signed_command :
+     signature_kind:Mina_signature_kind.t
+  -> Signed_command.t
+  -> ([ `Assuming of 'a list ], invalid) Result.t
+
+(** Collect verification key assumptions for zkApp proofs.
+
+    For each account update with proof authorization, collects the
+    (verification_key, statement, proof) tuple needed for async verification.
+
+    The return error type is open [[>]] to allow callers to extend with
+    additional error variants.
+
+    @return [Error invalid] if verification key is missing or mismatched *)
+val collect_vk_assumptions :
+     ('proof, _) verifiable_zkapp_command
+  -> ( (Side_loaded_verification_key.t * Zkapp_statement.t * 'proof) list
+     , [> `Missing_verification_key of
+          Signature_lib.Public_key.Compressed.t list
+       | `Unexpected_verification_key of
+         Signature_lib.Public_key.Compressed.t list ] )
+     Result.t
+
+(** Verify all signatures in a zkApp command.
+
+    {2 Verification Flow}
+
+    1. Compute transaction commitment from account updates
+    2. Compute full commitment (includes memo and fee payer)
+    3. Verify fee payer signature against full commitment
+    4. For each account update with signature authorization:
+       - Use full or partial commitment based on [use_full_commitment]
+       - Verify signature against [body.public_key]
+
+    This mirrors the logic enforced by the transaction SNARK
+    (see [Transaction_snark.check_authorization]).
+
+    @param signature_kind [Testnet] or [Mainnet] for domain separation *)
+val check_signatures_of_zkapp_command :
+     signature_kind:Mina_signature_kind.t
+  -> Zkapp_command.t
```

Comment:
This is interesting because it's more restrictive compared to the one defined in the module.